compute_sdk/tests/unit/test_client.py

```diff
@@ -411,76 +431,95 @@ def test_register_function_docstring(gcc, func):
     gcc.register_function(func)
     expected = inspect.getdoc(func)
 
-    a, _ = gcc._compute_web_client.v2.register_function.call_args
-    func_data = a[0]
-    assert func_data["description"] == expected
+    (data,), _ = gcc._compute_web_client.v3.register_function.call_args
 
-
-def test_register_function_no_metadata(gcc):
-    gcc._compute_web_client = mock.MagicMock()
-
-    gcc.register_function(funk)
-
-    a, _ = gcc._compute_web_client.v2.register_function.call_args
-    func_data = a[0]
-    assert func_data["metadata"] is None
-
-
-def test_register_function_no_function(gcc):
-    gcc._compute_web_client = mock.MagicMock()
-
-    with pytest.raises(ValueError) as pyt_exc:
-        gcc.register_function(None)
-
-    assert "either" in str(pyt_exc).lower()
-    assert "'function'" in str(pyt_exc).lower()
-    assert "'function_name'" in str(pyt_exc).lower()
-    assert "'function_code'" in str(pyt_exc).lower()
+    if expected:
+        assert data["description"] == expected
+    else:
+        assert "description" not in data
 
 
-def test_function_registration_data_function():
+def test_function_registration_data_data_function(serde):
```

Comment:
Is this different enough from test_register_function to justify both's existence?   the only diff is the function_code packed?

Reply:
For completeness and posterity, this tests a different (but related) unit.  That is, the `FunctionRegistrationData` class.  Note the test name and that `register_function` is not called.